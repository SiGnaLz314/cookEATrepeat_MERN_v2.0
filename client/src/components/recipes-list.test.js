import React from 'react';
import renderer from 'react-test-renderer';
import RecipesList from './recipes-list';
import { BrowserRouter } from 'react-router-dom';

const recipes=[
    {    recipe_id: 1,
        recipename: "Test",
        animal: "pork",
        ingredients: "Test",
        instructions:"Test",
        imagepath:"Test.jpg",
        imageURL: "https://cer-images.s3.amazonaws.com/images/Test.jpg",
        date: '2019-11-06T00:06:17.667+00:00',
    },
    {    recipe_id: 2,
        recipename: "Test",
        animal: "pork",
        ingredients: "Test",
        instructions:"Test",
        imagepath:"Test.jpg",
        imageURL: "https://cer-images.s3.amazonaws.com/images/Test.jpg",
        date: '2019-11-06T00:06:17.667+00:00',
    }
]; 
const adminFalse=false;
const loggedInFalse=false;
const adminTrue=true;
const loggedInTrue=true;


describe('Recipe-List tests', ()=> {
    test('Recipe-List snapshot test', () => {
        const component = renderer.create(
            <BrowserRouter>
                <RecipesList
                    recipes={recipes}
                    loggedIn={loggedInFalse}
                    admin={adminFalse} />
            </BrowserRouter>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Recipe-List snapshot Logged In & Admin test', () => {
        const component = renderer.create(
            <BrowserRouter>
                <RecipesList
                    recipes={recipes}
                    loggedIn={loggedInTrue}
                    admin={adminTrue} />
            </BrowserRouter>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
