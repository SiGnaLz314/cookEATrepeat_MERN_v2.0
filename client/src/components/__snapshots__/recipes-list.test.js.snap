// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe-List tests Recipe-List snapshot Logged In & Admin test 1`] = `
<div>
  <h3>
    cookEATrepeat Recipes
  </h3>
  <table
    className="table"
  >
    <thead
      className="thead-light"
    >
      <tr>
        <th>
          Recipe
        </th>
        <th>
          Animal
        </th>
        <th>
          Ingredients
        </th>
        <th>
          Instructions
        </th>
        <th>
          Date
        </th>
        <th>
          Recipe Image
        </th>
        <th>
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Test
        </td>
        <td>
          pork
        </td>
        <td>
          Test
          ...
        </td>
        <td>
          Test
          ...
        </td>
        <td>
          2019-11-06
        </td>
        <td>
          <div
            className="rec_img_box"
          >
            <a
              href="/recipe/1"
              onClick={[Function]}
            >
              <img
                alt="Not Available"
                id="recipe_img"
                src="https://cer-images.s3.amazonaws.com/images/Test.jpg"
                type="image"
              />
            </a>
          </div>
        </td>
        <td>
          <div
            id="recipe-action"
          >
            <button
              className="btn btn-outline-primary col-auto col-mr-auto"
            >
              <a
                href="/edit/1"
                onClick={[Function]}
              >
                edit
              </a>
            </button>
          </div>
          <div
            id="recipe-action"
          >
            <button
              className="btn btn-outline-primary col-auto col-mr-auto"
              href="#"
              onClick={[Function]}
            >
              <a
                href="/"
                onClick={[Function]}
              >
                delete
              </a>
            </button>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          Test
        </td>
        <td>
          pork
        </td>
        <td>
          Test
          ...
        </td>
        <td>
          Test
          ...
        </td>
        <td>
          2019-11-06
        </td>
        <td>
          <div
            className="rec_img_box"
          >
            <a
              href="/recipe/2"
              onClick={[Function]}
            >
              <img
                alt="Not Available"
                id="recipe_img"
                src="https://cer-images.s3.amazonaws.com/images/Test.jpg"
                type="image"
              />
            </a>
          </div>
        </td>
        <td>
          <div
            id="recipe-action"
          >
            <button
              className="btn btn-outline-primary col-auto col-mr-auto"
            >
              <a
                href="/edit/2"
                onClick={[Function]}
              >
                edit
              </a>
            </button>
          </div>
          <div
            id="recipe-action"
          >
            <button
              className="btn btn-outline-primary col-auto col-mr-auto"
              href="#"
              onClick={[Function]}
            >
              <a
                href="/"
                onClick={[Function]}
              >
                delete
              </a>
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Recipe-List tests Recipe-List snapshot test 1`] = `
<div>
  <h3>
    cookEATrepeat Recipes
  </h3>
  <table
    className="table"
  >
    <thead
      className="thead-light"
    >
      <tr>
        <th>
          Recipe
        </th>
        <th>
          Animal
        </th>
        <th>
          Ingredients
        </th>
        <th>
          Instructions
        </th>
        <th>
          Date
        </th>
        <th>
          Recipe Image
        </th>
        <th>
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Test
        </td>
        <td>
          pork
        </td>
        <td>
          Test
          ...
        </td>
        <td>
          Test
          ...
        </td>
        <td>
          2019-11-06
        </td>
        <td>
          <div
            className="rec_img_box"
          >
            <a
              href="/recipe/1"
              onClick={[Function]}
            >
              <img
                alt="Not Available"
                id="recipe_img"
                src="https://cer-images.s3.amazonaws.com/images/Test.jpg"
                type="image"
              />
            </a>
          </div>
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Test
        </td>
        <td>
          pork
        </td>
        <td>
          Test
          ...
        </td>
        <td>
          Test
          ...
        </td>
        <td>
          2019-11-06
        </td>
        <td>
          <div
            className="rec_img_box"
          >
            <a
              href="/recipe/2"
              onClick={[Function]}
            >
              <img
                alt="Not Available"
                id="recipe_img"
                src="https://cer-images.s3.amazonaws.com/images/Test.jpg"
                type="image"
              />
            </a>
          </div>
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
